spec/intro.dd

```diff
@@ -10,59 +10,47 @@ to create native libraries or executables.)
 $(P This document is the reference manual for the D Programming Language. For more information and
 other documents$(COMMA) see $(LINK2 https://dlang.org/, The D Language Website).)
 
-$(H2 Phases of Compilation)
-
-$(P The process of compiling is divided into multiple phases. Each phase has no
```

Comment:
I think this paragraph is useful, why is it being deleted?

Reply:
Hi @atilaneves, I explained this in the forum post: https://forum.dlang.org/post/pxllknizhivuvajlnhps@forum.dlang.org. I am copying the contents below:

```
I feel this description is unclear, and it might just reflect how DMD is implemented. I haven't implemented a C++ parser but parsers I have worked with - such as for C - it is always the case that lexer doesn't get impacted by the semantic analysis. The standard process is to get a stream of tokens from the lexer and work with that. It is also conceivable that someone could create a "dumb" AST first for C++, and as a subsequent phase add semantic meaning to the AST, just as is done for D.

For now I propose to remove this paragraph until there is a better description available.
```

I did request Walter for a better explanation of how D is different from other compilers but he hasn't yet responded. Once  a better description is available then I shall add it back.

Reply:
It definitly isn't the case for C++: `>>` could be a single token or two depending on whether it's the right shift operator or two closing template angle brackets.

Reply:
Hi @atilaneves so there is a parsing ambiguity in C++ but how does that imply that lexical phase is perturbed by later phases? There are multiple ways to handle this ambiguity : for instance lexer can return two '>' tokens rather than '>>' token and let the AST phase handle it.

Reply:
I am not sure I even understand what this is saying:

> Each phase has no dependence on subsequent phases. For example, the scanner is not perturbed by the semantic analyzer.

The C++ example you have given - how does it relate to this?

Reply:
I don't know the exact details, but I think that in C++ lexing can't be done independently of parsing.

Reply:
I am reluctant to keep the current wording as I don't know what it means. It would be clearer to state a) what kind of grammar D has (I asked about that but didn't get any answer), and b) something more concrete - such as: D's grammar is designed so that each lexical token unambiguously defines the production rule. 

I think that the way the ambiguity in C++ is handled in up to the implementation. It is possible to imagine an implementation that tokenizes '>>' as two '>' and lets a later phase handle '>>'. If  no one does it this way that doesn't make it impossible, it is likely it is less efficient to do it this way. 

In D too how are string mixins handled? Surely a mixin cannot be recognised until it is parsed, and then lexer has to be invoked on the mixin text? I see this dependency between phases.

Reply:
You'd have to ask Walter, he's the expert.

Reply:
Sure, asked already a couple of times on the forum, waiting for a reply.

The spec has to be technical and correct - we have to avoid vague or unsubstantiated claims, in particular if all they serve is to say D is better than X. The spec is not the place for comparisons; it just needs to describe the language with technical accuracy.

Reply:
> You'd have to ask Walter, he's the expert. 

You'd best email him.